File: frontend/src/app/mediconnect/models/Clinic.ts
import { Doctor } from "./Doctor";

export class Clinic 
{
    clinicId:number;
    clinicName:string;
    location:string;
    doctor: Doctor;
    contactNumber:string;
    establishedYear:number;
    constructor(clinicId:number,clinicName:string, location:string, contactNumber:string, establishedYear:number, doctor:Doctor,)
    {
        this.clinicId=clinicId;
        this.clinicName=clinicName;
        this.location=location;
        this.doctor=doctor;
        this.contactNumber=contactNumber;
        this.establishedYear=establishedYear;
    }
    // logAttributes()
    // {
    //     console.log('clinicId:', this.clinicId);
    //     console.log('clinicName:', this.clinicName);
    //     console.log('location:', this.location);
    //     console.log('doctorId:', this.doctor);
    //     console.log('contactNumber:', this.contactNumber);
    //     console.log('establishedYear:', this.establishedYear);
    // }
}

// export interface Clinic{
//     clinicId:number;
//     clinicName:string;
//     location:string;
//     doctor: Doctor;
//     contactNumber:string;
//     establishedYear:number;
// }

File: backend/src/main/java/com/wecp/progressive/entity/Clinic.java
package com.wecp.progressive.entity;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
public class Clinic {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int clinicId;
    @Column(name = "doctorId", insertable = false, updatable = false)
    private int doctorId;
    private String clinicName;
    private String location;
    // private int doctorId;
    private String contactNumber;
    private int establishedYear;

    @ManyToOne
    @JoinColumn(name = "doctorId")
    @JsonIgnore
    private Doctor doctor;

    public Clinic(int clinicId, String clinicName, String location, int doctorId, String contactNumber,
            int establishedYear) {
        this.clinicId = clinicId;
        this.clinicName = clinicName;
        this.location = location;
        this.doctorId = doctorId;
        this.contactNumber = contactNumber;
        this.establishedYear = establishedYear;
    }
    
    public Clinic() {
    }

    public int getClinicId() {
        return clinicId;
    }

    public void setClinicId(int clinicId) {
        this.clinicId = clinicId;
    }

    public String getClinicName() {
        return clinicName;
    }

    public void setClinicName(String clinicName) {
        this.clinicName = clinicName;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public int getDoctorId() {
        return doctorId;
    }

    public void setDoctorId(int doctorId) {
        this.doctorId = doctorId;
    }

    public String getContactNumber() {
        return contactNumber;
    }

    public void setContactNumber(String contactNumber) {
        this.contactNumber = contactNumber;
    }

    public int getEstablishedYear() {
        return establishedYear;
    }

    public void setEstablishedYear(int establishedYear) {
        this.establishedYear = establishedYear;
    }

    public Doctor getDoctor() {
        return doctor;
    }

    public void setDoctor(Doctor doctor) {
        this.doctor = doctor;
    }
    
}